apps/microtvm/arduino/template_project/src/example_project/platform.c

```diff
@@ -77,14 +88,19 @@ tvm_crt_error_t TVMPlatformTimerStop(double* elapsed_time_seconds) {
   return kTvmErrorNoError;
 }
 
-tvm_crt_error_t TVMPlatformGenerateRandom(uint8_t* buffer, size_t num_bytes) {
+// Fill a buffer with random data.
+__attribute__((weak)) tvm_crt_error_t TVMPlatformGenerateRandom(uint8_t* buffer, size_t num_bytes) {
```

Comment:
how come these are weak? if you miss implementing one, could you accidentally link the default and then wind up with two implementations?

Reply:
The idea here is that `platform.c` has one implementation of these functions which is added by default to the project. But in case the user wants to change the implementation, they could reimplement it in their `main.c`
That's why it is defined as weak.

Reply:
After discussing this with @alanmacd and @areusch, we decided to remove `weak` option since it is not widely supported in various compilers.